### PR TITLE
Add Mimecast TI Regional Data connector

### PR DESCRIPTION
**Pull request by lstajcic**
Fixes #

## Proposed Changes

  - Add Mimecast TI Regional Data connector

### CODE REVIEW

**Comment by lstajcic** [comment]
@v-jayakal NPM install is failing in most of the tests that used to pass. I can see that is happening to others also but just wanted to give a heads up about it in this PR also.

**Comment by lstajcic** [comment]
@v-jayakal I also need advice on what dataTypes field should look like for this Data connector. Data is submitted via https://docs.microsoft.com/en-us/graph/api/resources/tiindicator?view=graph-rest-beta so it is using ThreatIntelligenceIndicator table but this is Azure-Function Data connector. 

Thank you in advance.

**Review by NikTripathi — CHANGES_REQUESTED** [review]
Please add the zip folder.

**Comment by v-sabiraj** [comment]
Hey @lstajcic, We have resolved the issue for PR Validations failure and request you to please update your PR branch with the latest changes from **Azure:master** so that the PR Validations can be executed.... Thanks!!!

**Comment by v-sabiraj** [comment]
Hey @lstajcic, request you to please update your PR branch with the latest changes from Azure:master so that the PR Validations can be executed.Thanks.

File: Solutions/MimecastTIRegional/Data Connectors/MimecastTIRegional_API_AzureFunctionApp.json [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+  "publisher": "Mimecast",
+  "descriptionMarkdown": "The data connector for Mimecast Intelligence for Microsoft provides regional threat intelligence curated from Mimecast’s email inspection technologies with pre-created dashboards to allow analysts to view insight into email based threats, aid in incident correlation and reduce investigation response times.  \nMimecast products and features required: \n- Mimecast Secure Email Gateway \n- Mimecast Threat Intelligence\n",
+  "graphQueries": [],
+  "sampleQueries": [],
+  "dataTypes": [
+    {
+      "name": "Event(ThreatIntelligenceIndicator)"
+    }
+  ],
+  "connectivityCriterias": [],
```
Comment (NikTripathi) [inline Solutions/MimecastTIRegional/Data Connectors/MimecastTIRegional_API_AzureFunctionApp.json:13]: Connectivity criteria should not be empty.

**Review by NikTripathi — CHANGES_REQUESTED** [review]

**Comment by NikTripathi** [comment]
@lstajcic  Please add preview images for workbook. Thanks

**Comment by lstajcic** [comment]
> @lstajcic Please add preview images for workbook. Thanks

Where should I add those?

File: Solutions/MimecastTIRegional/Data Connectors/MimecastTIRegional_API_AzureFunctionApp.json (hunk 1 above)
Comment (lstajcic) [reply Solutions/MimecastTIRegional/Data Connectors/MimecastTIRegional_API_AzureFunctionApp.json]: @NikTripathi Done.

**Comment by NikTripathi** [comment]
> > @lstajcic Please add preview images for workbook. Thanks
> 
> Where should I add those?

Hi @lstajcic You can add preview images by creating a new folder named images in MimecastTIRegional folder and in that images folder create a folder preview and add those images there.

**Comment by lstajcic** [comment]
> 

Images uploaded.

**Comment by NikTripathi** [comment]
> Please add the zip folder.

@lstajcic please add the zip package.

File: Solutions/MimecastTIRegional/Data Connectors/MimecastTIRegional_API_AzureFunctionApp.json [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
...ecastAccessKey: Access Key for the dedicated Mimecast user\n - mimecastSecretKey: Secret Key for dedicated Mimecast user\n - mimecastBaseURL: Regional Mimecast API Base URL\n - activeDirectoryAppId: Azure portal ---> App registrations ---> [your_app] ---> Application ID\n - activeDirectoryAppSecret: Azure portal ---> App registrations ---> [your_app] ---> Certificates & secrets ---> [your_app_secret]\n\n >Note: If using Azure Key Vault secrets for any of the values above, use the`@Microsoft.KeyVault(SecretUri={Security Identifier})`schema in place of the string values. Refer to [Key Vault references documentation](https://docs.microsoft.com/azure/app-service/app-service-key-vault-references) for further details.\n\n4. Mark the checkbox labeled **I agree to the terms and conditions stated above**. \n5. Click **Purchase** to deploy.\n\n6. Go to ***Azure portal ---> Resource groups ---> [your_resource_group] --->  [appName](type: Storage account) ---> Storage Explorer ---> BLOB CONTAINERS ---> TIR checkpoints ---> Upload*** and create empty file on your machine named checkpoint.txt and select it for upload (this is done so that date_range for TIR logs is stored in consistent state)\n"
```
Comment (NikTripathi) [inline Solutions/MimecastTIRegional/Data Connectors/MimecastTIRegional_API_AzureFunctionApp.json:106]: Please update deploy to azure link as **https://aka.ms/sentinel-MimecastTIRegionalConnectorAzureFunctions-azuredeploy**

**Review by NikTripathi — CHANGES_REQUESTED** [review]

**Comment by NikTripathi** [comment]
@lstajcic please add the zip package.

File: Solutions/MimecastTIRegional/Data Connectors/MimecastTIRegional_API_AzureFunctionApp.json (hunk 2 above)
Comment (lstajcic) [reply Solutions/MimecastTIRegional/Data Connectors/MimecastTIRegional_API_AzureFunctionApp.json]: Done.

**Comment by NikTripathi** [comment]
> @lstajcic please add the zip package.

@lstajcic Zip package is still missing. Please update it.

**Comment by lstajcic** [comment]
@NikTripathi Done. pls check

**Comment by vmanojreddy** [comment]
Tested the data connector by deploying the functions. we see regex error for  function. Please check and update. Thanks.
image

**Comment by vmanojreddy** [comment]
@lstajcic please update the request changes. thanks!!!

**Comment by v-sabiraj** [comment]
Hey @lstajcic, we wanted to check on the status of this PR [Add Mimecast TI Regional Data connector](https://github.com/Azure/Azure-Sentinel/pull/3526#)
#3526. It is pending from more than 60 days. Let us know if any assistance is required for this PR. Per our standard operating procedures if no response is received in the next 7 business days we will close this PR. Thank you for your cooperation.

**Comment by v-sabiraj** [comment]
Hey @lstajcic, we wanted to check on the status of this PR [Add Mimecast TI Regional Data connector](https://github.com/Azure/Azure-Sentinel/pull/3526#)
#3526. It is pending from more than 60 days. Let us know if any assistance is required for this PR. Per our standard operating procedures if no response is received in the next 7 business days we will close this PR. Thank you for your cooperation.

**Comment by lstajcic** [comment]
@v-sabiraj Hello, I am sorry for this taking so long. I am on it, please don't close the PR.

**Comment by vmanojreddy** [comment]
@lstajcic Can you please provide estimate date for requested changes. thanks!!!

**Comment by vmanojreddy** [comment]
Hi @nkarisic please update the request changes. thanks!!!

**Comment by nkarisic** [comment]
> Hi @nkarisic please update the request changes. thanks!!!


@vmanojreddy j I have pushed all the requested changes, please let me know if anything else needs to be done, thanks!

**Comment by nkarisic** [comment]
> Hi @nkarisic please update the request changes. thanks!!!

Hi @vmanojreddy , i have updated the pull request with the requested changes, please review.

**Comment by nkarisic** [comment]
@vmanojreddy provided raw logo link, added .python_packages to data connector zip, please verify.

**Comment by vmanojreddy** [comment]
In Create ui file logo is not displaying properly. Can u please check once and fix it.
image

**Comment by v-dvedak** [comment]
Hey @lstajcic, @nkarisic, we wanted to check on the status of this PR https://github.com/Azure/Azure-Sentinel/pull/3526. It is pending from more than 60 days. Let us know if any assistance is required for this PR. Per our standard operating procedures if no response is received in the next 7 business days we will close this PR. Thank you for your cooperation.

**Comment by v-spadarthi** [comment]
Hi @lstajcic , since we have not received response from last 7 days, we are closing your PR as per our standard operating procedures. If you still need support for this issue, you can re-open the PR at any time.
If you do re-open, we simply request that you ensure the PR has response to the last request. Thank you for your cooperation.

**Pull request closed without merging**